File: 007.py
"""Find the 10,001st prime number."""

import math

PRIME_TO_FIND = 10001

def is_prime(n):
    """Check if n is a prime number."""
    for i in xrange(2, int(math.sqrt(n) + 1)):
        if n % i == 0:
            return False
    return True

i = PRIME_TO_FIND
j = 2
while i > 0:
    if is_prime(j):
        i -= 1
        if i == 0:
            print j
            break
    j += 1

File: 007.swift
#!/usr/bin/xcrun swift

import Foundation

// Find the 10,001st prime number.

let primeToFind = 10_001

func isPrime(n: Int) -> Bool {
    // Check if n is a prime number.
    for i in 2..<Int(sqrt(Double(n)) + 1) {
        if n % i == 0 {
            return false
        }
    }
    return true
}

var i = primeToFind
var j = 2
while i > 0 {
    if isPrime(j) {
        --i
        if i == 0 {
            println(j)
            break
        }
    }
    ++j
}
